app/registries/foreman/plugin.rb

```diff
@@ -342,14 +350,15 @@ def add_all_permissions_to_default_roles
       end
     end
 
-    def pending_migrations
+    def pending_migrations(context = ActiveRecord::MigrationContext.new(migrations_paths))
       return true if Foreman.in_setup_db_rake?
+      return @pending_migrations unless @pending_migrations.nil?
```

Comment:
This caching is a bit odd because it doesn't take `context` into account.

Reply:
what about now? I've taken the context out